### DISCUSSION TITLE
First try with instance segmentation: Issue with number of instances in prediction

### DISCUSSION THREAD

**Discussion by jeanpat** [PyTorchLightning/lightning-flash#1439] [Q&A]
Hello,
I finally to perform to run instance segmentation on
[a personal dataset (overlapping chromosome) in coco format](https://gist.github.com/jeanpat/5ba746d6cffb9abeb7ac451ee3ff8797).

The dataset consists in pairs of  overlapping chromosomes, that is in each image, there are two instances of the same class of object (chromosome). 
![image](https://user-images.githubusercontent.com/693574/187851392-6ad4c9f2-8ecc-4dfb-b033-bd42854a7886.png)

The model has two classes:

```
print(datamodule.labels)
print(datamodule.num_classes)
#help(datamodule.viz)
['background', 'chromosome']
2
```

When a prediction is performed, I get only one mask, for example:
![image](https://user-images.githubusercontent.com/693574/187852231-e632501d-44b5-4ae2-b467-ec222dbc837f.png)

I was waiting two masks. Does my dataset need to have three classes with background, chromosome1, chromosome2 ?

Thanks for your help

**Comment by ethanwharris** [comment]
Hi @jeanpat thanks for reporting this.

> Does my dataset need to have three classes with background, chromosome1, chromosome2 ?

I think so but not totally sure. Perhaps @Borda may have some insight from working with instance segmentation for this notebook: https://www.kaggle.com/code/jirkaborovec/cell-instance-segm-lightning-flash
